fake_switches/juniper/juniper_netconf_datastore.py

```diff
@@ -223,28 +225,33 @@ def parse_interfaces(self, conf, etree_conf):
         for interface_node in etree_conf.xpath("interfaces/interface/name/.."):
             handled_elements.append(interface_node)
 
-            port_name = val(interface_node, "name")
+            self.parse_interface(conf, interface_node)
 
-            port = conf.get_port_by_partial_name(port_name)
-            if port is None and re.match("^ae\d+$",port_name):
-                port = self.original_configuration.new("AggregatedPort", port_name)
-                port.vendor_specific["has-ethernet-switching"] = True
-                conf.add_port(port)
+        return handled_elements
 
-            self.raise_for_invalid_interface(port_name, self.MAX_AGGREGATED_ETHERNET_INTERFACES)
+    def parse_interface(self, conf, interface_node):
+        port_name = val(interface_node, "name")
+        self.raise_for_invalid_interface(port_name, self.MAX_AGGREGATED_ETHERNET_INTERFACES)
 
-            operation = resolve_operation(interface_node)
-            if operation in ("delete", "replace"):
-                backup = deepcopy(vars(port))
+        port = conf.get_port_by_partial_name(port_name)
+        if port is None and re.match("^ae\d+$", port_name):
```

Comment:
if the regex pattern good for all types of switches  or vendor-specific?

Reply:
it is good and is actually just a code refactor, not new code